components/raftstore/src/store/fsm/peer.rs

```diff
@@ -4913,6 +4955,12 @@ where
         let leader_id = self.fsm.peer.leader_id();
         let request = msg.get_requests();
 
+        // Forbid reads and writes when it's just became non-witness but not done apply
+        // snapshot.
+        if self.fsm.peer.wait_data {
```

Comment:
Shouldn't do it here. You should keep it from polling raft group like other peers applying snapshot. See `check_snap_status`

Reply:
I thought it wrong, forget about it.

Reply:
Please move it to L5010